Dockerfile

```diff
@@ -0,0 +1,78 @@
+FROM golang:1.13.1-stretch
+MAINTAINER Andrew Hill <andrew@textile.io>
+
+# This is (in large part) copied (with love) from
+# https://hub.docker.com/r/ipfs/go-ipfs/dockerfile
+
+# Get source
+ENV SRC_DIR /go-textile-threads
+
+# Download packages first so they can be cached.
+COPY go.mod go.sum $SRC_DIR/
+RUN cd $SRC_DIR \
+  && go mod download
+
+COPY . $SRC_DIR
+
+# build source
+RUN cd $SRC_DIR \
+  && go install github.com/textileio/go-textile-threads/daemon
+
+# Get su-exec, a very minimal tool for dropping privileges,
+# and tini, a very minimal init daemon for containers
+ENV SUEXEC_VERSION v0.2
+ENV TINI_VERSION v0.16.1
+RUN set -x \
+  && cd /tmp \
+  && git clone https://github.com/ncopa/su-exec.git \
+  && cd su-exec \
+  && git checkout -q $SUEXEC_VERSION \
+  && make \
+  && cd /tmp \
+  && wget -q -O tini https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini \
+  && chmod +x tini
+
+# Get the TLS CA certificates, they're not provided by busybox.
+RUN apt-get update && apt-get install -y ca-certificates
+
+# Now comes the actual target image, which aims to be as small as possible.
+FROM busybox:1-glibc
+MAINTAINER Sander Pick <sander@textile.io>
+
+# Get the threads binary, entrypoint script, and TLS CAs from the build container.
+ENV SRC_DIR /go-textile-threads
+COPY --from=0 /go/bin/daemon /usr/local/bin/threads
+COPY --from=0 $SRC_DIR/bin/container_daemon /usr/local/bin/start_threads
+COPY --from=0 /tmp/su-exec/su-exec /sbin/su-exec
+COPY --from=0 /tmp/tini /sbin/tini
+COPY --from=0 /etc/ssl/certs /etc/ssl/certs
+
+# This shared lib (part of glibc) doesn't seem to be included with busybox.
+# COPY --from=0 /lib/x86_64-linux-gnu/libdl-2.24.so /lib/libdl.so.2
+
+# Create the fs-repo directory
+ENV THREADS_REPO /data/threads
+RUN mkdir -p $THREADS_REPO \
+  && adduser -D -h $THREADS_REPO -u 1000 -G users textile \
+  && chown -R textile:users $THREADS_REPO
+
+# Switch to a non-privileged user
+USER textile
+
+# Expose the fs-repo as a volume.
+# start_threads initializes an fs-repo if none is mounted.
+# Important this happens after the USER directive so permission are correct.
+VOLUME $THREADS_REPO
+
+EXPOSE 4006
+EXPOSE 5050
+EXPOSE 9090
+EXPOSE 9091
+
+# This just makes sure that:
+# 1. There's an fs-repo, and initializes one if there isn't.
+# 2. The API and Gateway are accessible from outside the container.
+ENTRYPOINT ["/sbin/tini", "--", "/usr/local/bin/start_threads"]
+
+# Execute the daemon subcommand by default
+CMD ["daemon", "--repo=/data/threads", "--apiProxyAddr=/ip4/0.0.0.0/tcp/9091"]
```

Comment:
The install method above will put this under bin as `daemon` (we should change that so it installs as `threadsd`, if this is working, it must be using the container script differently, as in, not passing a _sub_ command.

Reply:
it does install as daemon, but then i move it to 'threads'. i ripped this from the other docker, where it will just be sure to ignore `daemon` if the user supplies it (or the default cmd does)

https://github.com/textileio/go-textile-threads/pull/169/files#diff-2d3d90a507605bc346c3abfa66979c19R17